### PR TITLE
Scenario for unnecessary slight right onto Holzmarktstr due to node offset

### PR DESCRIPTION
**Pull request by daniel-j-h**
Failing test for sensitive node placement. cc @MoKob 

# TODO
 - [x] discussion on how to handle node offsets
 - [x] check impact of https://github.com/Project-OSRM/osrm-backend/pull/2779 on this

## Requirements
 - [x] merge of https://github.com/Project-OSRM/osrm-backend/pull/2750 (base branch)
 - [x] merge of https://github.com/Project-OSRM/osrm-backend/pull/2727 (base branch)
 - [x] review
 - [x] adjust for comments

### CODE REVIEW

**Comment by MoKob** [comment]
I've managed to improve the perception here by considering the input/exit bearings in the collapse process. Due to the strong angle (for 10 meters) in the current scenario, the angle remains as a slight right, it emits a new-name now, though.

The new-name is technically correct for this setting. The test added here expects no instruction though. @daniel-j-h I feel that this might be an oversight in generating the test?

If we want to detect intense offsets like this one as straight, we need to look at ways to detect bad node-placement in the noisy data. I feel that this could be a bad idea though, as we would be fixing broken data at that point.

**Comment by daniel-j-h** [comment]
> The new-name is technically correct for this setting. The test added here expects no instruction though. @daniel-j-h I feel that this might be an oversight in generating the test?

Yes good catch.

File: src/engine/guidance/post_processing.cpp [hunk 1]
```diff
@@ -567,26 +568,150 @@ void collapseTurnAt(std::vector<RouteStep> &steps,
...
+                                       .bearings[current_step.intersections.front().in]),
+                               current_step.intersections.front()
+                                   .bearings[current_step.intersections.front().out]);
+
+                const auto bearing_turn_angle =
+                    turn_angle(util::bearing::reverseBearing(
+                                   one_back_step.intersections.front()
+                                       .bearings[one_back_step.intersections.front().in]),
+                               current_step.intersections.front()
+                                   .bearings[current_step.intersections.front().out]);
```
Comment (daniel-j-h) [inline src/engine/guidance/post_processing.cpp]: Want to write a small function abstracting over the common pattern here?

File: src/extractor/guidance/turn_analysis.cpp [hunk 2]
```diff
@@ -28,7 +28,7 @@ using EdgeData = util::NodeBasedDynamicGraph::EdgeData;
 
 bool requiresAnnouncement(const EdgeData &from, const EdgeData &to)
 {
-    return !from.IsCompatibleTo(to);
+    return !from.CanCombineWith(to);
```
Comment (daniel-j-h) [inline src/extractor/guidance/turn_analysis.cpp]: Can you explain this change to me?

Comment (MoKob) [inline src/extractor/guidance/turn_analysis.cpp]: It simply strengthens the check a bit.

**Pull request merged**